docs/dev_cycle.part-02.rst

```diff
@@ -0,0 +1,174 @@
+      return _process_result(sub_ctx.command.invoke(sub_ctx))
+    File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 895, in invoke
+      return ctx.invoke(self.callback, **ctx.params)
+    File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 535, in invoke
+      return callback(*args, **kwargs)
+    File "/usr/local/lib/python3.5/dist-packages/click/decorators.py", line 17, in new_func
+      return f(get_current_context(), *args, **kwargs)
+    File "/usr/local/lib/python3.5/dist-packages/populus/cli/compile_cmd.py", line 29, in compile_cmd
+      compile_project(project, watch)
+    File "/usr/local/lib/python3.5/dist-packages/populus/api/compile_contracts.py", line 18, in compile_project
+      _, compiled_contracts = compile_project_contracts(project)
+    File "/usr/local/lib/python3.5/dist-packages/populus/compilation/__init__.py", line 54, in compile_project_contracts
+      import_remappings=project.config.get('compilation.import_remappings'),
+    File "/usr/local/lib/python3.5/dist-packages/populus/compilation/backends/solc_auto.py", line 52, in get_compiled_contracts
+      return self.proxy_backend.get_compiled_contracts(*args, **kwargs)
+    File "/usr/local/lib/python3.5/dist-packages/populus/compilation/backends/solc_standard_json.py", line 131, in get_compiled_contracts
+      compilation_result = compile_standard(std_input, **command_line_options)
+    File "/usr/local/lib/python3.5/dist-packages/solc/main.py", line 184, in compile_standard
+      message=error_message,
+  solc.exceptions.SolcError: contracts/Donator.sol:21:5: DeclarationError: Identifier not found or not unique.
+  uin in_usd = msg.value * usd_rate;
+  ^-^
+  
+      > command: `solc --standard-json`
+      > return code: `0`
+      > stderr:
+      {"contracts":{},"errors":[{"component":"general","formattedMessage":"contracts/Donator.sol:21:5: DeclarationError: Identifier not found or not unique.\n    uin in_usd = msg.value * usd_rate;\n    ^-^\n","message":"Identifier not found or not unique.","severity":"error","type":"DeclarationError"}],"sources":{}}
+  
+      > stdout:
+
+What's that? actually it's not that bad. You can ignore the Python traceback, which is just the Populus call stack until the actual
+call to the compiler. 
+
+To undersatnd what went wrong, just look at the compiler's output. The error message is quite clear:
+
+.. code-block:: bash
+
+  solc.exceptions.SolcError: contracts/Donator.sol:21:5: DeclarationError: Identifier not found or not unique.
+  uin in_usd = msg.value * usd_rate;
+  ^-^
+  
+Oh. Ok. ``uin in_usd`` should be ``uint in_usd``. Edit and fix it:
+
+
+.. code-block:: bash
+
+  $ nano contracts/Donator.sol
+
+The fixed line should be:
+
+.. code-block:: solidity
+
+  uint in_usd = msg.value * usd_rate;
+  
+.. note::
+
+  Try the `online IDE <https://remix.ethereum.org>`_ , which has great interactive compiler and web-form like interface
```

Comment:
Change "online IDE" to "Remix IDE" so that it's referenced by name (people know it better as Remix)